app/furniture/marketplace/menu/product_component.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= render CardComponent.new(dom_id: dom_id(product)) do |card| %>
+  <%- card.with_header(variant: :no_padding) do %>
+    <% if product.photo.present? %>
+      <figure>
+        <%= image_tag hero_image, class: "rounded-t-lg w-full" %>
+        <figcaption class="px-2 pt-4 sm:px-4">
+          <h3><%= name %></h3>
+        </figcaption>
+      </figure>
+    <%- else %>
+      <h3 class="px-4"><%= name %></h3>
+    <% end %>
+  <%- end %>
+
+  <div class="text-sm italic">
+    <%= description %>
+  </div>
+
+  <div class="text-right mt-3">
+    <p><%= price %></p>
+  </div>
+
+  <%- card.with_footer do %>
+    <%- cart_product = cart.cart_products.find_by(product:) %>
+
+    <%- if !cart_product %>
+      <%= button_to("Add to Cart", cart.location(child: :cart_products), method: :post,  params: { cart_product: { product_id: product.id, quantity: 1 } }, class: "w-full --secondary") %>
+    <%- else %>
+      <%= render cart_product.quantity_picker %>
+    <%- end %>
```

Comment:
Could the first branch be part of the component. Would you ever not want to render these two branches together?

Reply:
We could, but we don't have a `quantity_picker` without a `cart_product` and I'm hesitant to instantiate a `cart_product` for every `product `on the page; because once you do it's in the in-memory cache that the `cart` has that `cart_product`; when it doesn't really.

Reply:
Ah ah right. I got a little confused that `QuantityPicker` is derived from `ActiveModel` and not `Application Component`. My suggestion works if the quantity picker is a View Component class with the `card_product` object passed in. Is `QuantityPicker` a view component in disguise? 😅

Reply:
It's definitely a `ViewComponent` in disguise. I tend to start from the `Model` layer and let things bubble out from there. For it to work as a `ViewComponent` it will need either the`cart` and `product` passed in or the `cart_product`. The main risk to mitigate is the "oops, we instantiated a new `CartProduct` associated with a `cart` and now it looks like we have more items in the cart  than we actually do!"

Which maybe I will do at some point but not today.